source/extensions/filters/common/lua/lua.h

```diff
@@ -60,6 +60,28 @@ namespace Lua {
  */
 #define DECLARE_LUA_CLOSURE(Class, Name) DECLARE_LUA_FUNCTION_EX(Class, Name, lua_upvalueindex(1))
 
+/**
+ * Calculate the maximum space needed to be aligned.
+ */
+template <typename T> constexpr size_t maximumSpaceNeededToAlign() {
+  return alignof(T) > alignof(void*) ? sizeof(T) : sizeof(T) + alignof(T) - 1;
```

Comment:
Should this be:

`return sizeof(T) + (alignof(T) % sizeof(T));` (or something like that?)

Do you have more details on the actual alignment problem that is happening? I'm a little confused as to what the JIT is doing here when user data is allocated.